docs/developer-docs/latest/update-migration-guides/migration-guides/v4/code/backend/controllers.md

```diff
@@ -0,0 +1,87 @@
+
+In Strapi v3, controllers export an object containing actions  that are merged with the existing actions of core API controllers, allowing customization.
+
+In Strapi v4, controllers export the result of a call to the [`createCoreController` factory function](/developer-docs/latest/development/backend-customization/controllers.md#implementation), with or without further customization.
+:::
+
+Migrating a controller to Strapi v4 consists in making sure that each controller is located in the proper folder and uses the `createCoreController` factory function introduced in v4.
+
+Due to the differences between controllers implementation in Strapi v3 and v4, it's recommended to create a new controller file, then optionally bring existing v3 customizations into the new file and adapt them when necessary.
+
+A new controller can be created:
+
+- manually, starting at step 1 of the following procedure,
+- or automatically, using the [`strapi generate` interactive CLI](/developer-docs/latest/developer-resources/cli/CLI.md#strapi-generate) then jumping directly to step 4 of the following procedure if controller customizations are required.
+
+To create a v4 controller:
+
+1. Create a `api/<api-name>/controllers/<controller-name>.js` file inside the `./src` folder (see [project structure](/developer-docs/latest/setup-deployment-guides/file-structure.md)).
+
+2. Copy and paste the following code at the top of the `./src/api/<api-name>/controllers/<controller-name>.js` file. The code imports the `createCoreController` factory function from the factories included with the core of Strapi:
+
+    ```js
+    const { createCoreController } = require('@strapi/strapi').factories;
+    ```
+
+3. Copy and paste the following code, replacing `api-name` and `content-type-name` with appropriate names. The code exports the result of a call to the `createCoreController` factory function, passing the unique identifier of the content-type (e.g. `api::api-name.content-type-name`) as an argument:
+
+    ```js
+    module.exports = createCoreController('api::api-name.content-type-name')
+    ```
+
+4. (_optional_) To customize controller actions, pass a second argument to the `createCoreController` factory function. This argument is an object containing methods, which can either be entirely new actions or replace or extend existing actions of core API controllers (see [controllers implementation documentation](/developer-docs/latest/development/backend-customization/controllers.md#adding-a-new-controller)).
+
+::: details Example of a v4 controller without customization
+
+  ```jsx
+  // path: ./src/api/<content-type-name>/controllers/<controller-name>.js
+
+  const { createCoreController } = require('@strapi/strapi').factories;
+
+  module.exports = createCoreController('api::api-name.content-type-name');
+  ```
+
+:::
+
+::: details Example of a v4 controller with customization
```

Comment:
I'd advise we have two examples here, one as a override (what we have currently) and a wrapper which is what we have in the documentation: 

![image](https://user-images.githubusercontent.com/8593673/152878653-ce1fdb25-89e7-4ab7-8de4-9812dbad1e94.png)


Wrapping / extending the controller is considered "more safe" in v4 with regards to upgrading versions in the future and is what should be recommended. Overriding should be a "last resort".

Reply:
+1. This way even functions like sanitizeOutput are only used when entirely necessary.

Reply:
Thank you, guys! I appreciate your help 😍 

I've [updated](https://github.com/strapi/documentation/pull/713/commits/e799a03b56cafde7e661ff983dc4c1c23139a7d9) the code example. The example wraps the original controller. Please feel free to edit it if you think something needs to be changed. 

I provided only one example because the migration guide's intent is not to give an exhaustive reference — the API reference is there for it, and we link it at least twice on the page.

But again feel free to suggest some code to illustrate what would be the safest way to customize a controller :-)